addons/pos_mercado_pago/static/src/app/payment_mercado_pago.js

```diff
@@ -0,0 +1,188 @@
+/** @odoo-module */
+import { _t } from "@web/core/l10n/translation";
+import { PaymentInterface } from "@point_of_sale/app/payment/payment_interface";
+import { ErrorPopup } from "@point_of_sale/app/errors/popups/error_popup";
+
+export class PaymentMercadoPago extends PaymentInterface {
+    setup() {
+        super.setup(...arguments);
+        this.webhook_resolver = null;
+        this.payment_cancel_ok_resolver = null;
+        this.payment_cancel_error_resolver = null;
+        this.payment_intent = {};
+        this.create_payment_intent = () =>
+            new Promise((resolve) => {
+                const line = this.pos.get_order().selected_paymentline;
+                // Build informations for creating a payment intend on Mercado Pago.
+                // Data in "external_reference" are send back with the webhook notification
+                const infos = {
+                    amount: line.amount * 100,
+                    additional_info: {
+                        external_reference: `${this.pos.pos_session.id}_${line.payment_method.id}`,
+                        print_on_terminal: true,
+                    },
+                };
+                // mp_payment_intent_create will call the Mercado Pago api
+                this.env.services.orm.silent
+                    .call("pos.payment.method", "mp_payment_intent_create", [
+                        [line.payment_method.id],
+                        infos,
+                    ])
+                    .then((result) => {
+                        resolve(result);
+                    });
+            });
```

Comment:
You can convert this into a method like so
```js
    async create_payment_intent() {
        const line = this.pos.get_order().selected_paymentline;
        // Build informations for creating a payment intend on Mercado Pago.
        // Data in "external_reference" are send back with the webhook notification
        const infos = {
            amount: line.amount * 100,
            additional_info: {
                external_reference: `${this.pos.pos_session.id}_${line.payment_method.id}`,
                print_on_terminal: true,
            },
        };
        // mp_payment_intent_create will call the Mercado Pago api
        const result = await this.env.services.orm.silent.call(
            "pos.payment.method",
            "mp_payment_intent_create",
            [[line.payment_method.id], infos]
        );

        return result;
    }
```

I suggest the same thing for `last_status_payment_intent` and `cancel_payment_intent`.

Reply:
https://github.com/odoo/odoo/pull/154962/commits/9eda4006837c31594bf32a16e164824a0f869fde